### ISSUE TITLE
ipadapter comfyui implementation with not-zeroed uncond image

### ISSUE THREAD

**Issue by cubiq** [tencent-ailab/IP-Adapter#77]
After the discussion in #51 I implemented a new node for ComfyUI that lets you add "noise" to the uncond image. I think the results are sometimes interesting and I'd love to hear what you guys think.

https://github.com/cubiq/ComfyUI_IPAdapter_plus

Also please note that this version uses less memory than the other one available for comfy as I return only the penultimate hidden states and not the whole tensor.

example:

![noise](https://i.imgur.com/DtgxRuk.jpg)

Basically a 0.01 noise adds a lot of noise making a very small impact on the result. A 1.0 noise removes noise so the impact is higher. When using noise >0 the checkpoint used has a greater effect on the image and it shows its character more so the actual textual prompt is more important.

thoughts? how would you create the noise? at the moment I'm still experimenting with it

**Comment by cubiq** [comment]
At first I did that. basically you could set a negative image but the results were a bit random and just adding more/less noise would yield to better images. I'll experiment more thanks for the suggestion. Maybe passing multiple images? Haven't checked the feasibility of that

**Comment by xiaohu2015** [comment]
> At first I did that. basically you could set a negative image but the results were a bit random and just adding more/less noise would yield to better images. I'll experiment more thanks for the suggestion. Maybe passing multiple images? Haven't checked the feasibility of that

One question, if we add noise to other images (not input image), how about the results?
by the way, I also added your code link in this repo (very thanks for your implementation)

**Comment by cubiq** [comment]
The base image is scrambled like so:

```
        TT.ElasticTransform(alpha=75.0, sigma=noise*3.5),
        TT.RandomVerticalFlip(p=1.0),
        TT.RandomHorizontalFlip(p=1.0),
```
on top of that I'm adding
```
image = image + ((0.25*(1-noise)+0.05) * torch.randn_like(image) )
```

so basically I'm not actually passing the original image, that would lead to very washed out results.

I use this technique so the base image becomes a sort of "seed" for the image generation as I'm using `torch.manual_seed(0)` for the noise itself so we can have repeatable results.

Ideally we want to send a set of "don't do this" negative images like they do here https://github.com/sd-fabric/fabric

PS: thanks for the link!

**Comment by xiaohu2015** [comment]
@cubiq got it.
